spring-xd-spark-streaming/src/main/java/org/springframework/xd/spark/streaming/examples/SparkLog.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.spark.streaming.examples;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.Properties;
+
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.api.java.function.VoidFunction;
+import org.apache.spark.streaming.api.java.JavaDStreamLike;
+
+import org.springframework.xd.spark.streaming.Processor;
+import org.springframework.xd.spark.streaming.SparkConfig;
+
+/**
+ * @author Mark Fisher
+ * @author Ilayaperumal Gopinathan
+ */
+@SuppressWarnings({ "unchecked", "rawtypes", "serial" })
+public class SparkLog implements Processor {
```

Comment:
The spark module implementation has to be inside the jar file as only jar files are included (sent across to spark cluster) as spark.jars when creating the spark streaming context.